x-pack/plugin/core/src/main/java/org/elasticsearch/xpack/core/ilm/CleanupTargetIndexStep.java

```diff
@@ -0,0 +1,129 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License
+ * 2.0; you may not use this file except in compliance with the Elastic License
+ * 2.0.
+ */
+package org.elasticsearch.xpack.core.ilm;
+
+import org.apache.logging.log4j.LogManager;
+import org.apache.logging.log4j.Logger;
+import org.elasticsearch.action.ActionListener;
+import org.elasticsearch.action.admin.indices.delete.DeleteIndexRequest;
+import org.elasticsearch.action.support.master.AcknowledgedResponse;
+import org.elasticsearch.client.internal.Client;
+import org.elasticsearch.cluster.ClusterState;
+import org.elasticsearch.cluster.metadata.IndexMetadata;
+import org.elasticsearch.common.Strings;
+import org.elasticsearch.core.TimeValue;
+import org.elasticsearch.index.IndexNotFoundException;
+
+import java.util.Objects;
+import java.util.function.Function;
+
+/**
+ * Deletes the target index created by an operation such as shrink or rollup and
+ * identified the target index name stored in the lifecycle state of the managed
+ * index (if any was generated)
+ */
+public class CleanupTargetIndexStep extends AsyncRetryDuringSnapshotActionStep {
```

Comment:
I can modify `ShrinkAction` so that it uses `CleanupTargetIndexStep` instead of `CleanupShrinkIndexStep` in a subsequent PR